source/Preprocessor.cpp

```diff
@@ -920,6 +923,11 @@ bool Preprocessor::expandMacro(DefineDirectiveSyntax* macro, Token usageSite, Ma
                 if (begin != end) {
                     dest.append(begin->withTrivia(alloc, token.trivia()));
                     dest.appendRange(++begin, end);
+                } else {
+                    // the macro argument contained no tokens.
+                    // we still need to supply an empty token here to ensure
+                    // that the trivia of the formal paremter is passed on
+                    dest.append(*alloc.emplace<Token>(TokenKind::EmptyMacroArgument, token.getInfo()));
```

Comment:
I don't understand the point of heap allocating and then immediately dereferencing.